external-crates/move/crates/move-unit-test/src/test_runner.rs

```diff
@@ -289,105 +303,219 @@ impl SharedTestingConfig {
+                for i in 0..self.num_iters {
+                    let mut iter_args = vec![];
+                    let prng_seed = self.prng_seed.unwrap_or(if self.deterministic_generation {
+                        i
+                    } else {
+                        rand::random::<u64>()
                     });
-                    let actual_err =
-                        MoveError(err.major_status(), sub_status, err.location().clone());
-                    assert!(err.major_status() != StatusCode::EXECUTED);
-                    match test_info.expected_failure.as_ref() {
-                        Some(ExpectedFailure::Expected) => {
-                            output.pass(function_name);
-                            stats.test_success(test_run_info, test_plan);
+                    let mut rng = StdRng::seed_from_u64(prng_seed);
+                    for arg in test_info.arguments.iter() {
+                        match arg {
+                            TestArgument::Value(v) => iter_args.push(v.clone()),
+                            TestArgument::Generate { generated_type } => {
+                                iter_args.push(Self::generate_value_for_typetag(
+                                    &mut rng,
+                                    generated_type,
+                                ));
+                            }
                         }
-                        Some(ExpectedFailure::ExpectedWithError(expected_err))
-                            if expected_err == &actual_err =>
-                        {
+                    }
+                    args.push((Some(prng_seed), iter_args));
+                }
+                args
+            };
+            let len = arguments.len();
+            for (i, (prng_seed, args)) in arguments.into_iter().enumerate() {
+                if !self.exec_test_once(
+                    test_plan,
+                    global_test_context,
+                    output,
+                    function_name,
+                    test_info,
+                    args,
+                    &mut stats,
+                    prng_seed,
+                    i == len - 1,
+                ) {
+                    break;
+                }
+            }
+        }
+
+        stats
+    }
+
+    fn generate_value_for_typetag(rng: &mut StdRng, ty: &TypeTag) -> MoveValue {
+        match ty {
+            TypeTag::Address => {
+                MoveValue::Address(AccountAddress::from_bytes(rng.gen::<[u8; 32]>()).unwrap())
+            }
+            TypeTag::U8 => MoveValue::U8(rng.gen::<u8>()),
+            TypeTag::U16 => MoveValue::U16(rng.gen::<u16>()),
+            TypeTag::U32 => MoveValue::U32(rng.gen::<u32>()),
+            TypeTag::U64 => MoveValue::U64(rng.gen::<u64>()),
+            TypeTag::U128 => MoveValue::U128(rng.gen::<u128>()),
+            TypeTag::U256 => MoveValue::U256(rng.gen::<U256>()),
+            TypeTag::Vector(ty) => {
+                let len = rng.gen_range(0..1024);
```

Comment:
We might want to make this configurable in some way in the future. Fine for now though

Reply:
Yea, totally agreed. I think this gets into "what extra args to `#[random_test]` to do we want etc. But definitely something to add further down the line.